Updates to advance API to version 0.4

geneorama opened the pull request:
Changes include:
 - updating examples to match date / time of specification
 - dropping alliance_id for patient_id to make API more general
 - minor formatting changes with spaces
 - minor formatting changes with breaks in ONC code table within table

geneorama committed:
fixed table (header area)

geneorama committed:
drafting more general version of patient_id based on alliance_id, updating spacing, adding breaks for ONC codes

geneorama committed:
updating date format in examples to reflect RFC 3339 standard

geneorama committed:
updating date format in examples to reflect RFC 3339 standard

geneorama commented:
@tomschenkjr I didn't see a place in the document to update the version number? Should we add that to the title or somewhere else within the document?

tomschenkjr commented on docs/index.md:
```diff
@@ -65,20 +65,20 @@ POST {json file} <url>/insert/
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}
+    {"date": "2017-07-25", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"},
```
Per the conversation with @geneorama and @avishekrk, this should be updated to _include_ timestamp, but not include subseconds.

tomschenkjr commented on docs/index.md:
```diff
@@ -15,43 +15,43 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| Field          | Format   | Constraints                                                | Notes/Questions                                                                                                   |
+|----------------|----------|------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| timestamp      | TimeDate | yyyy-mm-dd hh:mm:ss.sss-hh:mm                              | Current Timestamp. RFC 3339 (ISO 8601 derivative).                                                                |
+| clinic_id      | Text     |                                                            | Two Characted Organization ID (NN- NearNorth  EF- ErieFamily)                                                     |
```
We also need to include a `network_id` to identify the health network that is submitting data to us (e.g., Alliance, Northwestern, Presence Health, etc.).

tomschenkjr commented on docs/index.md:
```diff
@@ -15,43 +15,43 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+|----------------|----------|------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| timestamp      | TimeDate | yyyy-mm-dd hh:mm:ss.sss-hh:mm                              | Current Timestamp. RFC 3339 (ISO 8601 derivative).                                                                |
+| clinic_id      | Text     |                                                            | Two Characted Organization ID (NN- NearNorth  EF- ErieFamily)                                                     |
+| location_id    | Text     |                                                            | Specific Location Abbreviation in Centricity                                                                      |
```
@RaedMan is going to see if other networks also have a `clinic_id` and `location_id` equivalents. Namely, whether they have a concept of a clinic within their networks (e.g., a distinct sub-entity or sub-group within the network) and whether they also have a distinct location ID .

tomschenkjr reviewed (commented):
@geneorama - per the "version number", please make sure that references to the version number in the documentation is updated to "0.4.0". Likewise, look at the `./mkdocs.yml` file and update `site_name`. The site name appears in the main part of the navigation, so users will clearly see which version of the API is being used.

Finally, when we decide on v0.4.0, we will create a [GitHub release](https://help.github.com/articles/creating-releases/) so also denote it within the repo.

geneorama committed:
adding fields, making corrections, fixing more dates; updating specification for visit date, reverting date time for visit date example, updating version number, adding network_id, correcting alliance_id to patient_id, reordering response example to match documentation

geneorama commented:
@tomschenkjr Can you add Eric and Avishek so that I can reference them in the comments?

geneorama commented:
There is still a reference to Alliance:

| Field | Format | Constraints | Notes/Questions |
|-------|--------|-------------|-----------------|
| lab.route      | Text     | V  | *Always "V"  Venous for Alliance sites  |

@RaedMan or @potash would it be more appropriate to say

| Field | Format | Constraints | Notes/Questions |
|-------|--------|-------------|-----------------|
| lab.route      | Text     | V / C | "V"  Venous or "C" for Capillary |

tomschenkjr commented:
Will when I get a chance. You can still ref them though.

potash commented:
@geneorama Yes, Alliance only had venous tests but in general there will be capillary and venous. The codes in your update coincide with other CDPH lead databases (currbllshort and stellar).

Except for the fact that those databases use `sample_type` instead of `route` as the field name. We inherited `route` from Alliance.

tomschenkjr reviewed (changes requested):
In general, we need some more intentional description for each field. Right now, it's pretty tailored to Alliance's fields, but I think we're losing the intention behind each field. As evidenced by today's call, we sometimes forgot what each field is wanting. Let's make sure our notes are clearer to the intention, in addition to the structure that is being requested.

RaedMan commented:
@geneorama @potash Thanks! Yes, because of historic records it should be both, but capillary is being phased out in favor of venous because venous is reliable and less likely to be contaminated. The screening by C is cheap, fast, and easy, but the CDC states that C is presumptive and must be confirmed using V, but not sure how much C is being used in the model.  Thanks @potash for bring that up too because CDPH values are coming from IDPH and those are likely coming from the largest labs Quest and LabCorp and this is the EHR interoperability problem. @wesleycoates works with [HL7 standardization](http://www.hl7.org/index.cfm) but @tomschenkjr I'll also take the latest data dictionary to 3 hospitals and AHA and see what they have for the rest of the fields.

tomschenkjr commented:
So, seems like we can keep it as an item that is to either report V or C.

From: Raed Mansour [mailto:notifications@github.com]
Sent: Friday, September 08, 2017 5:29 PM
To: Chicago/lead-safe-api-docs <lead-safe-api-docs@noreply.github.com>
Cc: Schenk, Tom <Tom.Schenk@cityofchicago.org>; Mention <mention@noreply.github.com>
Subject: Re: [Chicago/lead-safe-api-docs] Updates to advance API to version 0.4 (#8)


@geneorama<https://github.com/geneorama> @potash<https://github.com/potash> Thanks! Yes, because of historic records it should be both, but capillary is being phased out in favor of venous because venous is reliable and less likely to be contaminated. The screening by C is cheap, fast, and easy, but the CDC states that C is presumptive and must be confirmed using V, but not sure how much C is being used in the model. Thanks @potash<https://github.com/potash> for bring that up too because CDPH values are coming from IDPH and those are likely coming from the largest labs Quest and LabCorp and this is the EHR interoperability problem. @wesleycoates<https://github.com/wesleycoates> works with HL7 standardization<http://www.hl7.org/index.cfm> but I'll also take the latest data dictionary to 3 hospitals and see what they have for the rest of the fields.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Chicago/lead-safe-api-docs/pull/8#issuecomment-328228677>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABkC0UP3A7i3PcT2SKVjPTzJOKDG_wvpks5sgb_GgaJpZM4PRjuA>.


________________________________
This e-mail, and any attachments thereto, is intended only for use by the addressee(s) named herein and may contain legally privileged and/or confidential information. If you are not the intended recipient of this e-mail (or the person responsible for delivering this document to the intended recipient), you are hereby notified that any dissemination, distribution, printing or copying of this e-mail, and any attachment thereto, is strictly prohibited. If you have received this e-mail in error, please respond to the individual sending the message, and permanently delete the original and any copy of any e-mail and printout thereof.

geneorama commented:
Here's my take on the discussion: It seems that there could be two types of tests, it's appropriate to ask the provider to abbreviate them as V or C, and `sample_type` is a more universal name for this field.

| Field | Format | Constraints | Notes/Questions |
|-------|--------|-------------|-----------------|
| lab.sample_type | Text     | V / C | "V"  Venous or "C" for Capillary |

geneorama committed:
fixing another alliance / patient_id reference

geneorama committed:
updating the Venus/Capillary field to remove Alliance reference and to be lab.sample_type

geneorama committed:
updating V/C description

geneorama commented:
We're trying to finalize the 0.4.0 version of the API today, and I would like to remind everyone to make any final questions today. @RaedMan @potash @avishekrk @wesleycoates 

The main goal here is to make the API more general and less Alliance specific. 

I have noticed a few more things: 
 - Within the Visit array we have `visit.location`, but within the main part of the record we've updated the location to be a combination of `network_id`, `clinic_id`, and `location_id`.  I'm assuming that we wouldn't need `network_id`, but shouldn't we need the clinic and location? @tomschenkjr thoughts on this?
 - In our analytics meeting we discussed the `visit.visit_id` which is described as "Unique GE ID for specific visit (DocumentID)". Although it's vendor specific, we decided that we shouldn't change this id. 
 - Same thing for `lab.lab_id` which is "Unique GE ID for specific lab result (ObsID)", that is staying the same. 

I was wondering on the last two if the description should say GE Centricity. 

Also, I was misspelling "Venous", there is no "i", thank you @potash for the correction.

RaedMan commented:
@geneorama I reached out on the varying EHRs but I will not have updates for everyone on standards by the end of today.

tomschenkjr commented:
@potash @avishekrk

Any other changes that you think are needed?

tomschenkjr commented on docs/index.md:
```diff
@@ -15,46 +15,48 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| network_id      | Text     |                                                            | The parent entity of the submitting organization                                                                  |
+| clinic_id       | Text     |                                                            | Two Characted Organization ID (NN- NearNorth  EF- ErieFamily)                                                     |
+| location_id     | Text     |                                                            | Specific Location Abbreviation in Centricity                                                                      |
+| patient_id      | Text     | AlphaNumeric                                               | A id to identify individual patients, should be unique with secondary key.                                        |
```
"seconday key"? Not sure I understand that.

geneorama replied:
For Alliance the `patient_id` is only unique within the context of a clinic / location. So one (or both) fields are needed to uniquely identify patients. 

The only reason we asked for the "location" was to uniquely identify patients. When we asked for the location we got two fields, `clinic_id` and `location_id`.  

I think I'll change the `patient_id` description to be simply "patient identifier", I think saying more is just confusing.

geneorama replied:
Also, the `location_id` is different than the `location` in the visit array.  The location in the visit array is just an address. As we discussed, I'll leave that alone.

geneorama replied:
I don't know what to do with the `clinic_id` field.  That seems to be very Alliance specific. I don't know how to modify that description to make sense for a broader audience of hospital groups. 

The `location_id` sounds more standard since that's a Centricity field.

tomschenkjr commented on docs/index.md:
```diff
@@ -109,24 +111,24 @@ POST {json file} <url>/insert/
 | version          | Text         | "0.2.1"                                                                         | Follows Semantic Versioning 2.0.0 http://semver.org/spec/v2.0.0.html                             |
 | timestamp        | TimeDate     | yyyy-mm-ddThh:mm:ss.sss-hh:mm                                                   | Current Timestamp. RFC 3339 (ISO 8601 derivative).                                               |
```
This is **still** inconsistent. With our timestamps, should there or should there not be a "T"?!

geneorama replied:
I made a number of changes based on your comments, and to make the documentation more consistent. I moved several "constraints" to "notes", because they were really notes. I also added `<N/A>` whenever there were no constraints to be clear that N/A doesn't mean "use an N or an A". 

Most of these changes are pretty clear, but there was one thing that I noticed for the first time that might be important; the visit array had a health center id which was not in the documentation.  I don't think we need that, I don't think we can expect that (or I know how to document it), so I removed it.  

I also reordered the array fields to match the documentation order to make it more clear / easier to check.

tomschenkjr commented on docs/index.md:
```diff
@@ -15,46 +15,48 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| Field           | Format   | Constraints                                                | Notes/Questions                                                                                                   |
+|-----------------|----------|------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| timestamp       | TimeDate | yyyy-mm-dd hh:mm:ss.sss-hh:mm                              | Current Timestamp. RFC 3339 (ISO 8601 derivative).                                                                |
+| visit_date      | Date     | yyyy-mm-dd                                                 | Effective date of score                                                                                           |
```
I don't see `visit_date` in the example?

geneorama replied:
adding...

tomschenkjr commented on docs/index.md:
```diff
@@ -15,46 +15,48 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+|-----------------|----------|------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| timestamp       | TimeDate | yyyy-mm-dd hh:mm:ss.sss-hh:mm                              | Current Timestamp. RFC 3339 (ISO 8601 derivative).                                                                |
+| visit_date      | Date     | yyyy-mm-dd                                                 | Effective date of score                                                                                           |
+| network_id      | Text     |                                                            | The parent entity of the submitting organization                                                                  |
```
Where is `network_id`?

geneorama replied:
adding...

tomschenkjr commented on docs/index.md:
```diff
@@ -65,20 +67,20 @@ POST {json file} <url>/insert/
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}
+    {"date": "2017-07-25 15:02:54.171-0:00", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"},
```
The documentation spec says three decimal points for the timestamp. This has two.

geneorama replied:
I think the decimals are there (that's the .171 part) but the 0:00 should be 00:00.

tomschenkjr commented on docs/index.md:
```diff
@@ -15,46 +15,48 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| Visit Array     |          |                                                            |                                                                                                                   |
+| visit.visit_id  | Numeric  | 16 Digit GE ID                                             | Unique GE ID for specific visit  (DocumentID)                                                                     |
+| visit.date      | Date     | yyyy-mm-dd hh:mm:ss.sss-hh:mm                              | RFC 3339 (ISO 8601 derivative).                                                                                   |
+| visit.location  | Text     |                                                            | Text Abbreviation for facility of visit location                                                                  |
```
This says that it should be the text abbreviation for the visit, but the example shows a fully specified address. Which is it?

tomschenkjr reviewed (changes requested):
There are a number of corrections and inconsistencies that remain. Please review carefully.

Moreover, the documentation is rather confusing and still very imprecise. Constraints and Notes/Questions seem to be interchangeable. While _we_ seem to understand what's happening, I think we still do get confused on what is expected in each field and what is permissible.

geneorama committed:
updates for 0.4.0 merge request

Commit:
Some tweaks to notes/description of fields

tomschenkjr commented:
I've made some small tweaks. @potash and @avishekrk, please also take a look.

geneorama commented:
Also, do we want to describe / define `etag`?

tomschenkjr commented:
@geneorama - I'm actually not sure what that is, yet. Can figure that out a bit later.

geneorama commented:
@tomschenkjr do we want to indicate that the `visit_date` is an optional parameter for submission? Just noticed you took that out.

tomschenkjr commented:
@geneorama - as of now, we haven't formally included "optional" parameters to the API so don't want to claim that without understanding the consequences.

tomschenkjr committed:
Merge branch 'master' into dev

geneorama commented:
@tomschenkjr ok, I'll update the  issue #7.  Based on my memory from the meeting on Friday I thought we were going to keep it optional, but I didn't remember anyone having clear opinions so it's hard to remember.

tomschenkjr commented:
We are staying away from optional at this time. There are very few elements here which are actually required and it would be possible to submit extra elements.

From: Gene Leynes [mailto:notifications@github.com]
Sent: Tuesday, September 12, 2017 2:27 PM
To: Chicago/lead-safe-api-docs <lead-safe-api-docs@noreply.github.com>
Cc: Schenk, Tom <Tom.Schenk@cityofchicago.org>; Mention <mention@noreply.github.com>
Subject: Re: [Chicago/lead-safe-api-docs] Updates to advance API to version 0.4 (#8)


@tomschenkjr<https://github.com/tomschenkjr> ok, I'll update the issue #7<https://github.com/Chicago/lead-safe-api-docs/issues/7>. Based on my memory from the meeting on Friday I thought we were going to keep it optional, but I didn't remember anyone having clear opinions so it's hard to remember.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Chicago/lead-safe-api-docs/pull/8#issuecomment-328957813>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABkC0Rrvy5vZBki00FNBqhMjRBTXrS9-ks5shtr6gaJpZM4PRjuA>.


________________________________
This e-mail, and any attachments thereto, is intended only for use by the addressee(s) named herein and may contain legally privileged and/or confidential information. If you are not the intended recipient of this e-mail (or the person responsible for delivering this document to the intended recipient), you are hereby notified that any dissemination, distribution, printing or copying of this e-mail, and any attachment thereto, is strictly prohibited. If you have received this e-mail in error, please respond to the individual sending the message, and permanently delete the original and any copy of any e-mail and printout thereof.

Commit:
Added column header to Header table

Commit:
Merge branch 'dev' of https://github.com/Chicago/lead-free-api-docs into dev

Commit:
Added Notes header to Response table

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
-| gender         | Text     | M/F/U                                                      |                                                                                                                                                                                                                             |
-| race           | Text     | Code Values from Centricity                                | Standard ONC Race definitionsCode     description1002-5  American Indian or Alaska Native2028-9  Asian  2054-5  Black or African American 2076-8  Native Hawaiian or Other Pacific Islander 2106-3  White UNK       Unknown |
-| ethnicity      | Text     | Code Values from Centricity                                | Standard ONC Ethnicity definitionsCode     description2135-2  Hispanic or Latino  2186-5  Non Hispanic or Latino UNK      Unknown                                                                                           |
-| Visit Array    |          |                                                            |                                                                                                                                                                                                                             |
```
I would clarify that this array may be empty. Should it be optional? In that case, missing means no visits. Ditto for the lab array.

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| gender          | Text     | M/F/U                              | Patient gender                                                                                                    |
+| race            | Text     | Code Values from Centricity        | Standard ONC Race definitions <br> <br> **Code / description** <br> 1002-5 / American Indian or Alaska Native <br> 2028-9 / Asian <br> 2054-5 / Black or African American <br> 2076-8 / Native Hawaiian or Other Pacific Islander <br> 2106-3 / White <br> UNK / Unknown |
+| ethnicity       | Text     | Code Values from Centricity        | Standard ONC Ethnicity definitions <br> <br> **Code / description** <br> 2135-2 / Hispanic or Latino <br> 2186-5 / Non Hispanic or Latino <br> UNK / Unknown |
+| VISIT ARRAY     |          |                                    |                                                                                                                   |
```
I would clarify that this array may be empty. Should it be optional? In that case, missing means no visits. Ditto for the lab array.

tomschenkjr replied:
Ok, this makes sense. We'll update to specify that it may be empty.

tomschenkjr replied:
@potash - from the perspective of the runner, which would you prefer to have: a `visit` array that is blank, i.e., `"visit": [ ]` or to just omit it overall? I'm assuming the latter.

tomschenkjr replied:
Per the discussion today, we will leave the entire array as optional (will do the same for `lab` array, too.

potash commented on docs/index.md:
```diff
@@ -65,76 +69,76 @@ POST {json file} <url>/insert/
+|------------------|--------------|-----------------------------------|--------------------------------------------------------------------------------------------------|
+| version          | Text         | "0.3.0"                           | Follows Semantic Versioning 2.0.0 http://semver.org/spec/v2.0.0.html                             |
+| timestamp        | TimeDate     | yyyy-mm-dd hh:mm:ss.sss-hh:mm     | Current Timestamp (RFC 3339 complient date).                                                     |
+| visit_date       | Date         | yyyy-mm-dd                        | Effective date of score                                                                          |
```
The reason I originally suggested it was because of both past and future visit dates. But:

1) past visit dates will, I believe, only be used internally for validation and that will probably happen by calling `predict.py` directly, no reason to do it over the API. if an external provider submits a past visit date, it seems like we'd want to handle that as an error.

2) future visit dates are plausible for providers, buta future visit date won't affect the returned prediction because we don't have future models trained on future data to use for the prediction, just the current one.

So not having a visit date makes things simpler. The only reason I can see for having one in the public API is to collect data on when patients are visiting, but we have that in the `visit` array. Sorry to go back on my original suggestion.

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| Field           | Format   | Constraints                        | Notes                                                                                                             |
+|-----------------|----------|------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| timestamp       | TimeDate | yyyy-mm-dd hh:mm:ss.sss-hh:mm      | Current timestamp (RFC 3339 complient date)                                                                       |
+| visit_date      | Date     | yyyy-mm-dd                         | Effective date of score                                                                                |
```
The reason I originally suggested it was because of both past and future visit dates. But:

1. past visit dates will, I believe, only be used internally for validation and that will probably happen by calling predict.py directly, no reason to do it over the API. if an external provider submits a past visit date, it seems like we'd want to handle that as an error.

2. future visit dates are plausible for providers, buta future visit date won't affect the returned prediction because we don't have future models trained on future data to use for the prediction, just the current one.

So not having a visit date makes things simpler. The only reason I can see for having one in the public API is to collect data on when patients are visiting, but we have that in the visit array. Sorry to go back on my original suggestion.

tomschenkjr replied:
Ok - we'll omit it.

potash commented on docs/index.md:
```diff
@@ -65,76 +69,76 @@ POST {json file} <url>/insert/
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}, 
-    {"date": "2017-07-25 15:02:54.171755", "visit_id": 1234567891011121, "location": "333 S State St", "provider": "John Doe"}
+    {"visit_id": 1234567891011121, "date": "2017-07-25 15:02:54.171-00:00", "location": "333 S State St", "provider": "John Doe"},
```
I'd suggest a shorter sample visit array-- no benefit from a long one and it takes up a lot of space on the page.

potash commented on docs/index.md:
```diff
@@ -65,76 +69,76 @@ POST {json file} <url>/insert/
-    {"lab_id": 1234567891011121, "route": "V", "healthcenterid": 1.0, "result": "13.6", "date": "2012-03-20", "type": "BLL"}, 
-    {"lab_id": 1234567891011121, "route": "V", "healthcenterid": 1.0, "result": "15", "date": "2012-02-21", "type": "BLL"}, 
-    {"lab_id": 1234567891011121, "route": "V", "healthcenterid": 1.0, "result": "5", "date": "2011-08-19", "type": "BLL"}] 
+    {"lab_id": 1234567891011121, "type": "BLL", "date": "2016-07-29", "sample_type": "V", "result": "None Detected ug/dL"}, 
```
Suggest a shorter lab array for the same reason as visit above.

tomschenkjr replied:
Agreed -- we've shortened to 3.

potash commented on docs/index.md:
```diff
@@ -65,76 +69,76 @@ POST {json file} <url>/insert/
+|------------------|--------------|-----------------------------------|--------------------------------------------------------------------------------------------------|
+| version          | Text         | "0.3.0"                           | Follows Semantic Versioning 2.0.0 http://semver.org/spec/v2.0.0.html                             |
+| timestamp        | TimeDate     | yyyy-mm-dd hh:mm:ss.sss-hh:mm     | Current Timestamp (RFC 3339 complient date).                                                     |
+| visit_date       | Date         | yyyy-mm-dd                        | Effective date of score                                                                          |
```
If `visit_date` is removed per my suggestion above, it should also be removed here.

tomschenkjr replied:
Makes sense -- also removing this.

potash commented on docs/index.md:
```diff
@@ -65,76 +69,76 @@ POST {json file} <url>/insert/
+| timestamp        | TimeDate     | yyyy-mm-dd hh:mm:ss.sss-hh:mm     | Current Timestamp (RFC 3339 complient date).                                                     |
+| visit_date       | Date         | yyyy-mm-dd                        | Effective date of score                                                                          |
+| patient_id       | Text         | AlphaNumeric                      | Same `patient_id` that was submitted to the API.                                                 |
+| risk_score       | AlphaNumeric | 9.99                              | Currently expecting numeric, future may be phrased by stating an odds ratio with the risk score. |
```
In my mind, alphanumeric does not include a decimal point so I would call this Text.

tomschenkjr replied:
Agreed.

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
 ### Submitting Prediction
-
+ 
 Submit a patient record and retrieved the estimated probability of having elevated blood-lead levels.
```
"retrieved" should be "retrieve". I'd replace "probability" with "risk" since we haven't decided that it will return a true probability.

tomschenkjr replied:
Agreed.

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
 ## API
 
 ### Submitting Prediction
```
I would change this because a prediction is not submitted. (A patient record is being submitted; and the prediction is retrieved.)

tomschenkjr replied:
Agreed

potash commented on docs/index.md:
```diff
@@ -15,46 +15,50 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
   "clinic_id": "EF",
   "location_id": "examp_loc",
-  "alliance_id": 9000, 
+  "patient_id": 9000, 
```
In the spec above `patient_id` is Text, but here is it numeric. I think Text is right (in order to support other EHRs that use text ids) so this should be "9000".

tomschenkjr replied:
Agreed. Likewise, we saw that `patient_id` in the response was also not in quotes. Great catch.

potash commented:
Not a change in this PR but this language

> provide estimates on a child's potential exposure to elevated lead levels

does not make sense to me because a child isn't exposed to elevated lead levels. Elevated blood lead levels occur as the result of exposure to lead hazards.

> estimate the likelihood that a child will have an elevated blood lead level

or 

> estimate the likelihood that a child will be exposed to lead-based paint hazards

potash commented:
In the credits: 
> The "Lead Safe" API was developed by the Chicago Department of Public Health, University of Chicago's Center for Data Science and Public Policy, and the Chicago Department of Innovation and Technology.

Please add my current affiliation:
> University of Chicago's [Center for Data Science and Public Policy](http://dsapp.uchicago.edu) and [Harris School of Public Policy](http://harris.uchicago.edu)

RaedMan commented:
@potash I prefer "estimate the likelihood that a child will be exposed to lead-based paint hazards" because it eliminates confusion that occurs with some that believe that the model also accounts for water and soil samples.

Commit:
Removed `visit_date`

Commit:
Shortened the number of examples in the array

Commit:
Modified language and fixed inconsistencies

Modified some of the language to be consistent and better-represent
the intent of the model. Also fixed inconsistencies in how patient_id
is suppose to be a string in the submission and response.

Commit:
Type-o fix

geneorama committed:
changed 'chance' to 'risk' to align more closely with other wording in the document

geneorama committed:
fixed spacing so that tables are easier to read

geneorama committed:
added escape to NA so that it prints

potash committed:
added my affiliation

geneorama commented:
I just made a few more edits, and split it between three commits.
 1. I changed the wording from "on the chance of" to "of the risk of" https://github.com/Chicago/lead-safe-api-docs/pull/8/commits/2db6b4d38cc86ba53d4af95d884242ddfad98c02
 2. Added and removed spaces so that the `|` symbols line up https://github.com/Chicago/lead-safe-api-docs/pull/8/commits/8236865daa894e94343713ab5efc96d3a8c25f71
 3. Changed `<N/A>` to `<N/A\>`, which allows the NA's to print normally using pymarkdown  2ebef3c4163b4c136180d4e501b1beb8b1bb311d

tomschenkjr commented on docs/index.md:
```diff
@@ -14,47 +14,49 @@ In order to integrate with Lead Safe, the health provider will need to make a RE
+| visit.date      | Date     | yyyy-mm-dd hh:mm:ss.sss-hh:mm      | Date of visit (RFC 3339 complient date)                                                                           |
+| visit.location  | Text     | <N/A\>                             | Text Abbreviation for facility of visit location                                                                  |
+| visit.provider  | Text     | <N/A\>                             | Full provider name, last name, or just NPI                                                                        |
+| LAB ARRAY       |          |                                    |                                                                                                                   |
```
After talking with the team today, we'll also have this as an optional item that does not need to be submitted.

tomschenkjr commented on .travis.yml:
```diff
@@ -0,0 +1,25 @@
+language: ruby
```
I committed the `.travis.yml` file by accident. I'll undo that.

tomschenkjr commented on docs/index.md:
```diff
@@ -65,76 +67,50 @@ POST {json file} <url>/insert/
+|------------------|--------------|-----------------------------------|--------------------------------------------------------------------------------------------------|
+| version          | Text         | "0.3.0"                           | Follows Semantic Versioning 2.0.0 http://semver.org/spec/v2.0.0.html                             |
+| timestamp        | TimeDate     | yyyy-mm-dd hh:mm:ss.sss-hh:mm     | Current Timestamp (RFC 3339 complient date).                                                     |
+| patient_id       | Text         | AlphaNumeric                      | Same `patient_id` that was submitted to the API.                                                 |
```
After talking with the team today, also need to add `network_id`, `clinic_id`, and `location_id` to the response document since it all needs to be present to make sense.

Commit:
Visit and Lab history now optional

Commit:
Added `network_id`, `clinic_id`, `location_id` to response document

Commit:
Didn't mean to commit travis file, undoing

Commit:
Tweaking format of submission table

tomschenkjr commented:
Self-approving these changes and accepting the merge request to `master`. Thanks, everyone!
